core/src/worker/data-feed.ts

```diff
@@ -549,6 +554,8 @@ export function deviationJob(reporterQueue: QueueType, _logger: Logger) {
       logger.debug({ aggregatorHash, fetchedAt: timestamp, submission }, 'Latest data aggregate')
       if (isStale({ timestamp, logger })) {
         logger.warn(`Data became stale (> ${MAX_DATA_STALENESS}). Not reporting.`)
+      } else if (!(await isRoundIdFresh(reporterQueue, oracleAddress, roundId))) {
+        logger.warn(`Data having low roundId:${roundId}. Not reporting`)
```

Comment:
If possible I would suggest to `return` after printing the warning to tell the reader that nothing important is happening afterward. It should be also in the `isStale` branch.